tests/server.mjs

```diff
@@ -1,52 +1,38 @@
-// Simple server adapted from https://developer.mozilla.org/en-US/docs/Learn/Server-side/Node_server_without_framework:
-import * as fs from "fs";
-import * as http from "http";
+// Simple server for local testing.
+
 import * as path from "path";
-const MIME_TYPES = {
-    default: "application/octet-stream",
-    html: "text/html; charset=UTF-8",
-    js: "application/javascript; charset=UTF-8",
-    mjs: "application/javascript; charset=UTF-8",
-    css: "text/css",
-    png: "image/png",
-    jpg: "image/jpg",
-    gif: "image/gif",
-    ico: "image/x-icon",
-    svg: "image/svg+xml",
-};
+import commandLineArgs from "command-line-args";
+import esMain from "es-main";
+import LocalWebServer from "local-web-server";
+
 
-const STATIC_PATH = path.join(process.cwd(), "./");
-const toBool = [() => true, () => false];
+const ROOT_DIR = path.join(process.cwd(), "./");
 
-export default function serve(port) {
+export default async function serve(port) {
     if (!port)
         throw new Error("Port is required");
-
-    const prepareFile = async (url) => {
-        const paths = [STATIC_PATH, url.pathname];
-        if (url.pathname.endsWith("/"))
-            paths.push("index.html");
-        const filePath = path.join(...paths);
-        const pathTraversal = !filePath.startsWith(STATIC_PATH);
-        const exists = await fs.promises.access(filePath).then(...toBool);
-        const found = !pathTraversal && exists;
-        const streamPath = found ? filePath : `${STATIC_PATH}/index.html`;
-        const ext = path.extname(streamPath).substring(1).toLowerCase();
-        const stream = fs.createReadStream(streamPath);
-        return { found, ext, stream };
+    const ws = await LocalWebServer.create({
+        port: port,
+        directory: ROOT_DIR,
+        corsOpenerPolicy: "same-origin",
+        corsEmbedderPolicy: "require-corp",
+    });
+    console.log(`Server started on http://localhost:${port}`);
```

Comment:
I noticed the log is output even though the server can't be started (for example run it twice in 2 terminals: the second one will return right away). I'm a bit surprised that `create` doesn't throw in that case...

Reply:
Maybe something such as this would work:
```js
ws.server.on('listening', () => console.log(`Server started on http://localhost:${port}`));
ws.server.on('error', (e) => console.error(`Error while starting the server`, e);
``

Reply:
And even our returned promise could wait for these events, and be either fulfilled or rejected depending on them, that would work better with the tests I believe.